app/controllers/offers_controller.rb

```diff
@@ -0,0 +1,108 @@
+class OffersController < ApplicationController
+  before_action :set_trade, only: [:new, :show, :create, :edit, :update, :accept]
+  before_action :set_offer, only: [:show, :edit, :update, :accept]
```

Comment:
Similarly to what Jonathan said, if you want the method to run on _most_ actions, you can do: `before_action :set_offer, except: [:new, :create]`  It's a judgment call about what expresses the intent better, and Rails can be both a blessing and a curse in the diversity of constructs it offers. :smile: